internal/jobqueue/jobqueue.go

```diff
@@ -0,0 +1,105 @@
+// Package jobqueue provides a generic interface to a simple job queue.
+//
+// Jobs are pushed to the queue with Enqueue(). Workers call Dequeue() to
+// receive a job and FinishJob() to report one as finished.
+//
+// Each job has a type and arguments corresponding to this type. These are
+// opaque to the job queue, but it mandates that the arguments must be
+// serializable to JSON. Similarly, a job's result has opaque result arguments
+// that are determined by its type.
+//
+// A job can have dependencies. It is not run until all its dependencies have
+// finished.
+package jobqueue
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"time"
+
+	"github.com/google/uuid"
+)
+
+// JobQueue is an interface to a simple job queue. It is safe for concurrent use.
+type JobQueue interface {
```

Comment:
Should there also be a `cancelled` method? That can obviously be a follow-up, but we discussed it, so just making sure I understood the intentions correctly.

Reply:
There should be indeed. Also the ability for jobs to time out. I wanted to get this in with the minimal functionality and address those as follow-ups, because this change is already big as it is.

Reply:
Makes perfect sense to me.